### ISSUE TITLE
Issue with the PS2 controller

### ISSUE THREAD

**Issue by nschuetz1311** [intel/ecfw-zephyr#38]
I encountered an issue with the PS2 controller when pressing  CTRL + ALT + Shift + "any key".

This leads to the EC crashing 
```
[00:01:03.481,170] <err> os: ***** USAGE FAULT *****
[00:01:03.481,201] <err> os:   Attempt to execute undefined instruction
[00:01:03.481,231] <err> os: r0/a1:  0x00000013  r1/a2:  0x001197f9  r2/a3:  0x00000001
[00:01:03.481,231] <err> os: r3/a4:  0x000002f9 r12/ip:  0x00000000 r14/lr:  0x000e15d9
[00:01:03.481,262] <err> os:  xpsr:  0x61000074
[00:01:03.481,262] <err> os: Faulting instruction address (r15/pc): 0x000e1620
[00:01:03.481,292] <err> os: >>> ZEPHYR FATAL ERROR 0: CPU exception on CPU 0
[00:01:03.481,323] <err> os: Fault during interrupt handling

[00:01:03.481,323] <err> os: Current thread: 0x118fe8 (unknown)
[00:01:03.487,457] <err> os: Halting system
```

Is this a known issue, if so is there a way to fix this?

**Comment by albertofloyd** [comment]
@nschuetz1311 This is not a known issue, could you provide more details? 

1) Board are you building the project
2) Any Kconfig/device tree customizations
3) Full UART log prior to exception with following log configuration.
   west build -c -p always -b **<BOARD>** -- -DOVERLAY_CONFIG="debug.conf" -DCONFIG_THREAD_NAME=y  
4) List file (build/zephyr/zephyr.lst  so we can determine point of exception?

**Comment by nschuetz1311** [comment]
The board i tested this on is a ADL-S RVP (IOTG DDR5 1DPC) .

The codebase is zephyr 3.1 currently.

I tested activating  "CONFIG_EC_FUJITSU_KEYBOARD" and "CONFIG_EC_GTECH_KEYBOARD" as I guess this issue somehow is related to the "keymap_get_fnkey". The EC did not crash with these options but the Key combination was still not transmitted to the host system.

One of our customers is using a PS2 Keyboard their system with photoshop an tried to use the before mentioned key combination.
We tried this with an adjusted EC FW, which at least did not crash the EC but still did not process the key combination correctly.

Can you process these keys correctly on your side in the OS (preferably a Windows operating system)?

**Comment by albertofloyd** [comment]
> The board i tested this on is a ADL-S RVP (IOTG DDR5 1DPC) .
So this is the HW used, do confirm you are building mec1501_adl and not mec1501_adl_p

> The codebase is zephyr 3.1 currently.
Looks like you have a customized baseline, there is no basic EC open against Zephyr 3.1.

> 
> I tested activating "CONFIG_EC_FUJITSU_KEYBOARD" and "CONFIG_EC_GTECH_KEYBOARD" as I guess this issue somehow is related to the "keymap_get_fnkey". The EC did not crash with these options but the Key combination was still not transmitted to the host system.
> 
This doesn't make sense, since these options are for a physical keyboard matrix not for PS2. 
As requested do describe any further customizations

> One of our customers is using a PS2 Keyboard their system with photoshop an tried to use the before mentioned key combination. We tried this with an adjusted EC FW, which at least did not crash the EC but still did not process the key combination correctly.
> 
> Can you process these keys correctly on your side in the OS (preferably a Windows operating system)?
Yes, this was verified during the last release.

**Comment by nschuetz1311** [comment]
So I tested it again on an unmodified Intel BIOS (last test was on an AMI BIOS) the crash was a porting error on my side.

However, the mentioned key combination does not work as expected.
We tried it with PhotoShop and also with the On-Screen-Keyboard in the OS (Windows 11).
The only combinations working are "ctrl" + "alt" or "shift" + "alt" or "ctrl" + "shift", but not all three of them.

The board I tested this on is an RPL-S ADP-S DDR5 UDIM CRB with project Version: RPLSFWI.1.R00. 3087.A01.2202231211 and EC-Version: 01.51
